models/user.go

```diff
@@ -520,5 +520,49 @@ func (u *User) IsBanned() bool {
 
 func (u *User) UpdateBannedUntil(tx *storage.Connection) error {
 	return tx.UpdateOnly(u, "banned_until")
+}
+
+// RemovePassword removes the user's password and email provider information
+func (u *User) RemovePassword(tx *storage.Connection) error {
```

Comment:
maybe we should rename this to `RemoveUnconfirmedIdentities`? we will also need to check if a user's account has a phone identity and remove it